changelog/next/features/5123--read_lines-split-regex.md

```diff
@@ -0,0 +1,2 @@
+We added a `split_regex` option to allow the use of regular expressions to
+split events with the `read_lines` operator.
```

Comment:
Can you add a few tests for this? Ideally something like this under `tenzir/tests/exec/*`?

```
from (
  {chunk: "foo--"},
  {chunk: "---bar-"},
  {chunk: "qu"},
  {chunk: "x-"},
)
write_lines
read_lines split_regex="-*"
```